algebra-core/src/serialize/mod.rs

```diff
@@ -100,32 +103,42 @@ pub trait CanonicalDeserialize: Sized {
     }
 }
 
-impl CanonicalSerialize for u64 {
-    #[inline]
-    fn serialize<W: Write>(&self, writer: &mut W) -> Result<(), SerializationError> {
-        Ok(writer.write_all(&self.to_le_bytes())?)
-    }
+macro_rules! impl_uint {
+    ($ty: ident) => {
+        impl CanonicalSerialize for $ty {
+            #[inline]
+            fn serialize<W: Write>(&self, writer: &mut W) -> Result<(), SerializationError> {
+                Ok(writer.write_all(&self.to_le_bytes())?)
+            }
 
-    #[inline]
-    fn serialized_size(&self) -> usize {
-        Self::SERIALIZED_SIZE
-    }
-}
+            #[inline]
+            fn serialized_size(&self) -> usize {
+                Self::SERIALIZED_SIZE
+            }
+        }
 
-impl ConstantSerializedSize for u64 {
-    const SERIALIZED_SIZE: usize = 8;
-    const UNCOMPRESSED_SIZE: usize = Self::SERIALIZED_SIZE;
-}
+        impl ConstantSerializedSize for $ty {
+            const SERIALIZED_SIZE: usize = core::mem::size_of::<$ty>();
+            const UNCOMPRESSED_SIZE: usize = Self::SERIALIZED_SIZE;
+        }
 
-impl CanonicalDeserialize for u64 {
-    #[inline]
-    fn deserialize<R: Read>(reader: &mut R) -> Result<Self, SerializationError> {
-        let mut bytes = [0u8; 8];
-        reader.read_exact(&mut bytes)?;
-        Ok(u64::from_le_bytes(bytes))
-    }
+        impl CanonicalDeserialize for $ty {
+            #[inline]
+            fn deserialize<R: Read>(reader: &mut R) -> Result<Self, SerializationError> {
+                let mut bytes = [0u8; Self::SERIALIZED_SIZE];
+                reader.read_exact(&mut bytes)?;
+                Ok($ty::from_le_bytes(bytes))
+            }
+        }
+    };
 }
 
+impl_uint!(u8);
+impl_uint!(u16);
+impl_uint!(u32);
+impl_uint!(u64);
+impl_uint!(usize);
```

Comment:
Hmm because this can change depending on architecture, we might not want to impl this (unless we serialize `usize`s as `u64`s always)